docs/connection/connection-spans.md

```diff
@@ -0,0 +1,180 @@
+
+**[1]:** It's REQUIRED to document error types instrumentation produces. It's RECOMMENDED to use error codes provided by the socket library, runtime, or the OS (such as `connect` method error codes on [Linux or other POSIX systems](https://man7.org/linux/man-pages/man2/connect.2.html#ERRORS) or [Windows](https://docs.microsoft.com/windows/win32/api/winsock2/nf-winsock2-connect#return-value)).
+
+**[2]:** If and only if a connection (attempt) ended with an error.
+
+**[3]:** The value SHOULD be normalized to lowercase.
+
+Consider always setting the transport when setting a port number, since
+a port number is ambiguous without knowing the transport. For example
+different processes could be listening on TCP port 12345 and UDP port 12345.
+
+**[4]:** The value SHOULD be normalized to lowercase.
+
+**[5]:** When observed from the client side, and when communicating through an intermediary, `server.address` SHOULD represent the server address behind any intermediaries, for example proxies, if it's available.
+
+`error.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `_OTHER` | A fallback error value to be used when the instrumentation doesn't define a custom value. |
+
+`network.transport` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `tcp` | TCP |
+| `udp` | UDP |
+| `pipe` | Named or anonymous pipe. |
+| `unix` | Unix domain socket |
+
+`network.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `ipv4` | IPv4 |
+| `ipv6` | IPv6 |
+<!-- endsemconv -->
+
+## Examples
+
+### Successful connection
+
+Successful connection attempt to `"/tmp/my.sock"` results in the following span:
+
+|   Attribute name       |         Value       |
+| :--------------------- | :-------------------|
+| name                   | `"connect"`         |
+| `network.peer.address` | `"/tmp/my.sock"`    |
+| `network.transport`    | `"unix"`            |
+
+Once corresponding connection is gracefully closed, another span is reported:
+
+|   Attribute name       |         Value       |
+| :--------------------- | :-------------------|
+| name                   | `"connection"`      |
+| `network.peer.address` | `"/tmp/my.sock"`    |
+| `network.transport`    | `"unix"`            |
+
+### Successful connect, but connection terminates with an error
+
+Successful connection attempt to `example.com` results in the following span:
+> Note: DNS lookup is outside of the scope of this semantic convention
```

Comment:
While I agree that we shouldn't be trying to include all the dns info in the connection - having an event/marker for timing that indicates when it was complete would be helpful.
If dns is being tracked with its own spans - having a convention for linking from the connection span to dns would make sense.